src/components/app/index.js

```diff
@@ -45,22 +45,43 @@ class App extends React.Component {
 
 App.propTypes = {
   data: PropTypes.oneOfType([
-    PropTypes.string,
+    PropTypes.oneOf(['random', 'lorem', 'animals', 'demo', 'json']),
     PropTypes.shape({
-      created_ts: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+      schema_id: PropTypes.string,
       edges: PropTypes.array.isRequired,
-      message: PropTypes.string,
       nodes: PropTypes.array.isRequired,
       tags: PropTypes.array
     })
-  ])
+  ]),
+  theme: PropTypes.oneOf(['dark', 'light']),
+  visible: PropTypes.shape({
+    labelBtn: PropTypes.bool,
+    themeBtn: PropTypes.bool
+  })
```

Comment:
Would providing a config object be more flexible than baking the names into this prop types e.g.

```
config: {
   visibility: {
       labelBtn: false
   }
}
```

With this you could create a HOC to show / hide it e.g.

```
function withVisibility(WrappedComponent, config, componentName) {
  // ...and returns another component...
  return class extends React.Component {
    render() {
      if (typeof config.visibility[componentName] === 'boolean') {
          if (!config.visibility[componentName]) return null
      }
      return <WrappedComponent {...this.props} />;
    }
  };
}
```

You then use it like

``` 
const LabelButton = ({ onToggle, textLabels }) => (
  <button
    aria-live="polite"
    className={classnames('pipeline-toggle-labels pipeline-icon-button', {
      'pipeline-toggle-theme--show-labels': textLabels
    })}
    onClick={() => onToggle(!textLabels)}>
    <span>{textLabels ? 'Hide' : 'Show'} text labels</span>
    <LabelIcon className="pipeline-icon" />
  </button>
);

export withVisibility(LabelButton, config, 'labelBtn')
```

The reason this might something to explore is that if hiding parts of the application is needed more and more it becomes trivial to enable this functionality for any component, rather than having lots of guarded checks dotted around the application (like in src/components/icon-toolbar/index.js)

Reply:
Nice idea, but a little too clever for my taste. This smells like premature/unnecessary abstraction to me. I'm trying to keep things simple and relatively accessible for non-expert devs.